delphi_srr2_radar_driver_wrapper/src/delphi_srr2_radar_driver_wrapper.cpp

```diff
@@ -78,13 +80,13 @@ void DelphiSrr2RadarDriverWrapper::status5_cb(const delphi_srr_msgs::SrrStatus5C
 }
 
 void DelphiSrr2RadarDriverWrapper::detection_cb(const radar_msgs::RadarDetectionArrayConstPtr &msg) {
-    last_update_time_ = ros::Time::now();
     track_msg_ = msg;
 }
 
 void DelphiSrr2RadarDriverWrapper::publish_object_track()
 {
-    if(track_msg_ == nullptr)
+    if(track_msg_ == nullptr ||
+       ros::Time::now() - track_msg_->header.stamp > ros::Duration(msg_timeout_))
```

Comment:
A timeout will likely still result in duplicate messages. Since this driver is single threaded, it probably makes more sense to just create a prev_track_msg_ and check if track_msg_ == prev_track_msg 
You could do the same for the status message.

Also since all he subscriber call backs simply assign to a variable, there is no need for them to have queues of size 10. It could be 1. Only one message will be processed for publication in this paradigm.

Reply:
That makes sense. I guess I can also use the sequence number to check message duplication. In that case, I do not have to keep two copies of radar track message.

Reply:
That could probably work as well, but it does introduce a dependency on seq num

Reply:
Yeah... Probably I will just go in your way.